### PR TITLE
fix(specmap): Do not resolve $refs in most example values

### PR DESCRIPTION
**Pull request by hkosova**
### Description
This PR is a continuation of https://github.com/swagger-api/swagger-js/pull/1465. It updates the $ref resolver to prevent resolution inside example values where references are not supported.

### Motivation and Context
Fixes https://github.com/swagger-api/swagger-ui/issues/4765 and https://github.com/swagger-api/swagger-ui/issues/5625.

### How Has This Been Tested?
Added tests + tested manually using a custom build of Swagger Editor.

### Screenshots (if appropriate):
image

### Types of changes

- [ ] No code changes (changes to documentation, CI, metadata, etc)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by char0n** [comment]
Thanks for your work @hkosova! I'll provide a review during monday.

File: test/specmap/refs.js [hunk 1]
```diff
@@ -467,6 +508,86 @@ describe('refs', () => {
...
+      test('should ignore root or nested $refs in values of OAS3 media type `examples`', () => {
+        const input = {
+          openapi: '3.0.0',
+          paths: {
+            '/': {
+              post: {
+                requestBody: {
+                  content: {
+                    'application/json': {
+                      schema: {},
```
Comment (char0n) [inline test/specmap/refs.js:528]: Can we add `example` entry here alongside with `examples` to have test for https://github.com/swagger-api/swagger-js/pull/1810/files#diff-649819c985ad609b5b71b21fc77464a381e6e202379c30f480dab171a19b250fR43 ?

**Review by char0n — APPROVED** [review]
Great job @hkosova! I've cross-referenced all the current paths with fixtures used in tests and I found one possible test function that is not covered with test/fixtures. Can you please look into it (left a comment there)? Very nice work.

File: test/specmap/refs.js (hunk 1 above)
Comment (hkosova) [reply test/specmap/refs.js:528]: @char0n the test for `content.*.example` already exists here:
https://github.com/swagger-api/swagger-js/blob/master/test/bugs/ui-4924.js

Should I move this test to `test/specmap/refs.js` for consistency?

**Comment by hkosova** [comment]
please build

File: test/specmap/refs.js (hunk 1 above)
Comment (char0n) [reply test/specmap/refs.js:528]: @hkosova yes, please. We're moving away from test fragmentation in `test/bugs/` directory.

Comment (hkosova) [reply test/specmap/refs.js:586]: ✔️  done

**Pull request merged**